inst/rmarkdown/templates/transmissibility/skeleton/skeleton.Rmd

```diff
@@ -378,46 +376,42 @@ discretized Gamma.
 
 ## Results
 
-```{r, eval = params$use_epiparameter}
-si_epiparameter <- epiparameter::epidist(
-  params$epiparameter_pathogen,
-  "serial_interval"
+```{r, eval = params$use_epiparameter_database}
+si_epidist <- epiparameter::epidist_db(
+  disease =  params$disease_name,
+  epi_dist =  "serial_interval",
+  single_epidist = TRUE,
+  subset = is_parameterised
 )
-si_params <- si_epiparameter$param
-si_dist <- si_epiparameter$dist
-si_mean <- si_params[[1]]
-si_sd <- si_params[[2]]
+
+si_params <- get_parameters(si_epidist)
+si_dist <- family(si_epidist)
+si_mean <- get_parameters(si_epidist)[1]
+si_sd <- get_parameters(si_epidist)[2]
 ```
 
-```{r, eval = !params$use_epiparameter}
+```{r, eval = !params$use_epiparameter_database}
 si_mean <- params$si_mean
 si_sd <- params$si_sd
 si_dist <- params$si_dist
 si_cv <- si_sd / si_mean
-si_params <- epitrix::gamma_mucv2shapescale(mu = si_mean, cv = si_cv)
+si_epidist <- epidist(
+  disease = params$disease_name,
+  epi_dist = "serial_interval",
+  prob_distribution = params$si_dist,
+  summary_stats = create_epidist_summary_stats(mean = params$si_mean,
+                                               sd = params$si_sd),
+  auto_calc_params = TRUE)
 ```
 
 ```{r}
-# do.call() because we want to pass `si_params` names as the name of arguments.
-# This is important if params in si_params are not the first arguments of the
-# q...() function. For example, if a gamma distribution is define by its shape
-# and scale, instead of shape and rate.
-si <- do.call(
-  distcrete,
-  c(
-    list(si_dist,
-      interval = 1,
-      w = 0
-    ),
-    si_params
-  )
-)
-si_x <- seq(1L, to = si$q(0.999), by = 1L)
+si <- epiparameter::discretise(si_epidist)
```

Comment:
You could replace this line with using `discretise = TRUE` in the previous `epidist()` call, which should produce the same output. It personal preference really. If they do not produce the same output let me know.

Reply:
so the thing here is that the distribution to discretise can either come from epiparameter or from the user directly, that's why this step is included here separately